engine/etcdng/v3/etcd.go

```diff
@@ -615,12 +615,14 @@ func (n *ng) Subscribe(changes chan interface{}, afterIdx uint64, cancelC chan s
 
 	for response := range watchChan {
 		if response.Canceled {
-			log.Infof("stop watching: graceful shutdown")
-			return nil
-		}
-		if err := response.Err(); err != nil {
-			log.Errorf("stop watching: error: %v", err)
-			return err
+			log.Warn("etcd watcher cancelled")
+
+			if err := response.Err(); err != nil {
```

Comment:
In practice I've found that we can get an error without `Cancelled = true`. So we should check for it separately, but log it if it's set when `Cancelled = true`. (etcd watchers are in such a horrible state)

Reply:
ok :+1: